pkg/apis/servicecatalog/validation/clusterid_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+func TestValidateClusterID(t *testing.T) {
+	cases := []struct {
+		name  string
+		id    *sc.ClusterID
+		valid bool
+	}{
+		{
+			"happy path, name is cluster-id, ID is set",
+			&sc.ClusterID{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster-id",
+				},
+				ID: "some-id",
+			},
+			true,
+		},
+		{
+			"missing ID",
+			&sc.ClusterID{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster-id",
+				},
+			},
+			false,
```

Comment:
I think this shouldn't be an error case. If they don't provide some unique ID let's create one for them. The caller may not easily know how (or want to) generate their own UUID, and in that case I don't see any reason not to make it easier on them and do it for them.  Yes,we should still allow them to specify one if they have a specific value they want to use.

Reply:
We can create one for them using defaults. However, at this point in the code path, the defaults have already been applied. The validator should definitely reject a missing ID, and this test is testing the validator.

Reply:
My comment may not have been in the right spot in the code - its just when it popped into my head :-)